connection.go

```diff
@@ -106,10 +98,7 @@ func (c *Connection) Close() error {
 // This function is used internally by Run which should be used for most queries.
 func (c *Connection) Query(q Query) (*Response, *Cursor, error) {
 	c.mu.Lock()
-	if c == nil {
```

Comment:
This is good to go and handled [https://github.com/nerdylikeme/gorethink/commit/c5660df8a18dae23bacd36cb74cfdf736831f6c6](here).